### PR TITLE
ARROW-87: [C++] Add all four possible ways to encode Decimals in Parquet to schema conversion

### PR DESCRIPTION
**Pull request by xhochy**
See also: https://github.com/Parquet/parquet-format/blob/master/LogicalTypes.md#decimal

### CODE REVIEW

File: cpp/src/arrow/parquet/parquet-schema-test.cc [hunk 1]
```diff
@@ -117,6 +119,40 @@ TEST_F(TestConvertParquetSchema, ParquetFlatPrimitives) {
...
+  parquet_fields.push_back(
+      PrimitiveNode::Make("int32-decimal", Repetition::OPTIONAL,
+          parquet_cpp::Type::INT32,
+          parquet_cpp::LogicalType::DECIMAL, -1, 8, 4));
+  arrow_fields.push_back(std::make_shared<Field>("int32-decimal", DECIMAL_8_4));
+
+  parquet_fields.push_back(
+      PrimitiveNode::Make("int64-decimal", Repetition::OPTIONAL,
+          parquet_cpp::Type::INT64,
+          parquet_cpp::LogicalType::DECIMAL, -1, 8, 4));
```
Comment (wesm) [inline cpp/src/arrow/parquet/parquet-schema-test.cc]: Aside, it might be nice to improve this API in parquet-cpp -- the `-1` is a bit of an eyesore

**Comment by wesm** [comment]
Looks good. I can merge after you rebase -- we need to get the Parquet tests running in Travis CI. We might look into caching Ubuntu 14.04 library artifacts someplace for parquet-cpp to cut down on build times (e.g. building Thrift takes a little while in the Parquet builds)

**Comment by xhochy** [comment]
Rebased.

**Comment by wesm** [comment]
Thank you. +1

**Pull request closed without merging**